From 3aa900bf793ce4598aa3274ea11d1d85e7deec92 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 3 Sep 2012 18:11:22 +0900
Subject: [PATCH 52/97] dfb-video-example: add create_video_pipeline function.

Group video pipeline creation into a new function, create_video_pipeline().
---
 ext/directfb/dfb-video-example.c |   14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/ext/directfb/dfb-video-example.c b/ext/directfb/dfb-video-example.c
index 0a9675d..6391dd0 100644
--- a/ext/directfb/dfb-video-example.c
+++ b/ext/directfb/dfb-video-example.c
@@ -41,7 +41,7 @@ usage (char *cmd)
 }
 
 static void
-on_pad_added (GstElement * element, GstPad * pad, gpointer data)
+create_video_pipeline (GstPad * pad, gpointer data)
 {
   GstPad *sinkpad;
   GstElement *peer_element = (GstElement *) data;
@@ -50,9 +50,6 @@ on_pad_added (GstElement * element, GstPad * pad, gpointer data)
   const gchar *mime;
   static GstElement *decoder, *parser;
 
-  /* We can now link this pad with the gst-omx decoder or h264parse sink pad */
-  printf ("Dynamic pad created, linking\n");
-
   if (decoder) {
     /* If decoder plugin has been already created, demuxer plugin just gets
        a link to a peer pad. This route is passed when the state is changed. */
@@ -119,6 +116,15 @@ on_pad_added (GstElement * element, GstPad * pad, gpointer data)
 }
 
 static void
+on_pad_added (GstElement * element, GstPad * pad, gpointer data)
+{
+  /* We can now link this pad with the gst-omx decoder or h264parse sink pad */
+  printf ("Dynamic pad created, linking\n");
+
+  create_video_pipeline (pad, data);
+}
+
+static void
 play_handler (int signum)
 {
   switch (signum) {
-- 
1.7.10.4

